pkg/kubelet/kubelet.go

```diff
@@ -969,7 +975,42 @@ func makeMounts(container *api.Container, podVolumes kubecontainer.VolumeMap) (m
 			ReadOnly:      mount.ReadOnly,
 		})
 	}
-	return
+	if mountEtcHostsFile {
+		hostsMount, err := makeHostsMount(kl.getPodDir(pod.UID), pod.Status.PodIP, pod.Name, container.Name)
+		if err != nil {
+			return nil, err
+		}
+		mounts = append(mounts, *hostsMount)
+	}
+	return mounts, nil
+}
+
+func makeHostsMount(podDir, podIP, podName, containerName string) (*kubecontainer.Mount, error) {
+	hostsFilePath := path.Join(podDir, fmt.Sprintf("%s-etc-hosts", containerName))
```

Comment:
we should not need a file per container - can we make the hosts file once after the pod is created and then use that for all of the container mounts?